src/lib/core/datetime.c

```diff
@@ -163,9 +163,12 @@ tm_to_datetime(struct tnt_tm *tm, struct datetime *date)
 size_t
 datetime_strptime(struct datetime *date, const char *buf, const char *fmt)
 {
-	assert(date != NULL);
-	assert(fmt != NULL);
-	assert(buf != NULL);
+	if ((date == NULL) ||
+	    (fmt == NULL) ||
+	    (buf == NULL)) {
+		return -1;
```

Comment:
And we return 0, not -1 (as it's length of parsed prefix) if there are some premature errors.

Reply:
replaced with modified version of assert